packages/transaction-controller/src/utils/batch.ts

```diff
@@ -410,14 +412,25 @@ async function addTransactionBatchWithHook(
   const collectHook = new CollectPublishHook(transactionCount);
   try {
     if (requireApproval && useHook) {
+      const { gasLimit, transactions: transactionsWithGas } =
```

Comment:
Should we also check `isSimulationEnabled` and either throw or fallback using our feature flag values?